### PR TITLE
FIX Update Travis configuration to be standalone, add codecov.io

### PR DESCRIPTION
**Pull request by robbieaverill**

### CODE REVIEW

**Comment by robbieaverill** [comment]
I need to fix that fixtured theme that's failing

File: .travis.yml [hunk 1]
```diff
@@ -1,28 +1,32 @@
...
-  - cd ~/builds/ss
-  - composer install
+  - phpenv rehash
+  - phpenv config-rm xdebug.ini
+
+  - composer install --prefer-dist
+  - composer require --prefer-dist --no-update silverstripe/framework:4.0.x-dev
+  - composer update
+
+  - if [[ $DB == PGSQL ]]; then composer require silverstripe/postgresql:2.0.x-dev --prefer-dist; fi
```
Comment (dhensby) [inline .travis.yml]: Move this able the `composer update` line and add `--no-update` for faster overall installs

File: composer.json [hunk 2]
```diff
@@ -14,7 +14,8 @@
         "silverstripe/cms": "^4.0@dev"
     },
     "require-dev": {
-        "phpunit/PHPUnit": "~4.8"
+        "phpunit/phpunit": "^5.7",
+        "squizlabs/php_codesniffer": "*"
```
Comment (dhensby) [inline composer.json]: Use a semver constraint here

File: src/Model/Widget.php [hunk 3]
```diff
@@ -260,15 +260,15 @@ public function getController()
...
+        foreach (array_reverse(ClassInfo::ancestry(get_class($this))) as $widgetClass) {
             $controllerClass = "{$widgetClass}Controller";
             if (class_exists($controllerClass)) {
                 break;
             }
         }
 
         if (!class_exists($controllerClass)) {
-            throw new Exception("Could not find controller class for $this->classname");
+            throw new Exception("Could not find controller class for get_class($this)name");
```
Comment (fspringveldt) [inline src/Model/Widget.php]: Don't think this will work as the classname won't get printed.

Comment (robbieaverill) [reply src/Model/Widget.php]: Ha, that'll teach me for search/replacing

**Pull request merged**